firefox-ios/Client/Frontend/Browser/Toolbars/AddressToolbarContainer.swift

```diff
@@ -426,4 +418,22 @@ final class AddressToolbarContainer: UIView,
             store.dispatch(action)
         }
     }
+
+    private func applyProgressBarTheme(isPrivateMode: Bool, theme: Theme) {
+        let gradientStartColor = isPrivateMode ? theme.colors.borderAccentPrivate : theme.colors.borderAccent
+        let gradientMiddleColor = isPrivateMode ? nil : theme.colors.iconAccentPink
+        let gradientEndColor = isPrivateMode ? theme.colors.borderAccentPrivate : theme.colors.iconAccentYellow
+
+        progressBar.setGradientColors(
+            startColor: gradientStartColor,
+            middleColor: gradientMiddleColor,
+            endColor: gradientEndColor
+        )
+    }
+}
+
+extension AddressToolbarContainer: PrivateModeUI {
+    func applyUIMode(isPrivate: Bool, theme: Theme) {
+        applyProgressBarTheme(isPrivateMode: isPrivate, theme: theme)
```

Comment:
@thatswinnie I have a stash with changes going through redux flow but I couldn't find an action where applyTheme had `isPrivateMode` updated with the value. If this feel too damage control we could pair with your knowledge of Toolbar we could make it work

Reply:
@yoanarios I don't see an issue using `applyUIMode` here. However we should not use an `extension` here as we wanted to move away from using them. So have the class conform to the protocol `PrivateModeUI` directly.